internal/webdav/fs.go

```diff
@@ -0,0 +1,102 @@
+package webdav
+
+import (
+	"context"
+	"os"
+	"path"
+	"strings"
+
+	"bazil.org/fuse"
+	fusefs "bazil.org/fuse/fs"
+	"github.com/restic/restic/internal/debug"
+	"golang.org/x/net/webdav"
+)
+
+// RepoFileSystem implements a read-only file system on top of a repositoy.
+type RepoFileSystem struct {
+	ctx  context.Context
+	root fuseDir
+}
+
+// Mkdir creates a new directory, it is not available for RepoFileSystem.
+func (fs *RepoFileSystem) Mkdir(ctx context.Context, name string, perm os.FileMode) error {
+	return webdav.ErrForbidden
```

Comment:
I'm not sure this error is supposed to be used by client code of x/net/webdav: its description says something about webdav.LockSystem. Returning something like errors.New("modifying a repository is not possible") would be cleaner, I think.